components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Content.java

```diff
@@ -177,6 +179,22 @@ public static void applyToAsset(final Asset asset, final AttributesMap contentAt
   }
 
   /**
+   * Finds fresh {@link Asset} instance from passed in TX by entity ID of the {@link Asset} used
+   * to create passed in {@link Content} instance. The passed in {@link Content} must have been created with
+   * method {@link #extractFromAsset(Asset, Iterable, AttributesMap)} to have proper attributes set for this operation.
+   *
+   * @see #extractFromAsset(Asset, Iterable, AttributesMap)
+   */
+  @Nullable
+  public static Asset findAsset(final StorageTx tx, Content content) {
```

Comment:
Righto, if its use is coupled with `extractFromAsset` then it makes sense to leave it where it is.